### ISSUE TITLE
`importNames` and static call with imported class causes `RemoveExtraParametersRector`

### ISSUE THREAD

**Issue by jackbentley** [rectorphp/rector#7794]
# Bug Report

I'm unsure how to replicate this. However, after looking over the Rector code I can provide an explanation and a potential fix.

The code in question is a static method call to an imported class:

```php
use App\My\Event;

// ...

Event::add('a', 'b', 'c', 'd');
```

What happens is when Rector imports fully qualified class names via the `importNames` flag, parameters on the `Event::add` call removed. 

For example, this (in theory) would cause the issue to appear. I've been unable to replicate with the online "Try Rector" tool.

```diff
  use App\My\Event;
+ use Some\Fully\Qualified\ClassName;
  
- function fooBar() : \Some\Fully\Qualified\ClassName
+ function fooBar() : ClassName
{
-     Event::add('a', 'b', 'c', 'd');
+     Event::add('a');
}
```

From what I can gather debugging the Rector code, it looks this line returns a reflection for a class inside of `phpstan.phar`. Specifically from this file `phpstan.phar/vendor/ondrejmirtes/better-reflection/src/SourceLocator/SourceStubber/../../../../../jetbrains/phpstorm-stubs/event/event.stub`

https://github.com/rectorphp/rector-src/blob/f9f4217a6d0cf0a13f17af52a4c4f52e246fbe5b/rules/Php71/Rector/FuncCall/RemoveExtraParametersRector.php#L69

There is a check for this kind of thing happening in `shouldSkipFunctionReflection` but it's only for function calls and not classes.

Weirdly this class if in the global namespace and the class usage in the code is in a namespace (via an import).

Further debugging revealed that the value of `$objectType` when resolving the static call into a reflection is of the type `ShortenedObjectType`. This explains how the the class ended up being resolved to the global namespace because (as I understand) the method `getClassName` should return the fully qualified class name and this object returns the short name.

https://github.com/rectorphp/rector-src/blob/main/src/Reflection/ReflectionResolver.php#L138

I can fix my specific issue by adding a check in `shouldSkipFunctionReflection` for the filename of a declared class or by adding a check for `ShortenedObjectType` in `resolveMethodReflectionFromStaticCall`. However as I don't understand the code fully, it feels like there might be a wider issue here.

**TL;DR:** `ShortenedObjectType` causes issues with reflections/resolve of a class name.

**Comment by TomasVotruba** [comment]
Fixed in https://github.com/rectorphp/rector-src/pull/3397
